prql-compiler/src/parser/expr.rs

```diff
@@ -0,0 +1,233 @@
+use std::collections::HashMap;
+
+use chumsky::prelude::*;
+use itertools::Itertools;
+
+use super::common::*;
+use super::lexer::{InterpolItem, Token};
+use crate::ast::pl::*;
+
+pub fn expr_call() -> impl Parser<Token, Expr, Error = Simple<Token>> {
+    func_call(expr())
+}
+
+pub fn expr() -> impl Parser<Token, Expr, Error = Simple<Token>> + Clone {
+    recursive(|expr| {
+        let literal = select! { Token::Literal(lit) => lit }.map(ExprKind::Literal);
+
+        let ident_kind = ident().map(ExprKind::Ident);
+
+        let list = ctrl("[")
+            .ignore_then(
+                ident_part()
+                    .then_ignore(ctrl("=").padded_by(whitespace().or_not()))
+                    .or_not()
+                    .then(func_call(expr.clone()))
+                    .map(|(alias, expr)| Expr { alias, ..expr })
+                    .padded_by(whitespace().or(new_line()).repeated())
+                    .separated_by(ctrl(","))
+                    .allow_trailing(),
+            )
+            .then_ignore(whitespace().or(new_line()).repeated())
+            .then_ignore(ctrl("]"))
+            .map(ExprKind::List)
+            .labelled("list");
+
+        let pipeline = ctrl("(")
+            .ignore_then(pipeline(func_call(expr)))
+            .then_ignore(ctrl(")"));
```

Comment:
Can we use `delimited_by` for these?

Reply:
It seems so :D

Reply:
Wait, no `delimited_by` is for things that are iterable (I may be reading the error wrong).

Reply:
Hmmm, I'll try this out; atm I'm getting an error building

Reply:
In general, `a.ignore_then(inner).then_ignore(c)` can be replaced with `inner.delimited_by(a, b)`.